src/features/home/components/AllEventsList.tsx

```diff
@@ -1,17 +1,152 @@
-import { FC } from 'react';
-import { Box } from '@mui/material';
+import { FC, useState } from 'react';
+import {
+  Avatar,
+  Box,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Switch,
+} from '@mui/material';
+import { KeyboardArrowDown } from '@mui/icons-material';
 
+import { getContrastColor } from 'utils/colorUtils';
 import useAllEvents from 'features/events/hooks/useAllEvents';
 import EventListItem from './EventListItem';
+import useMemberships from 'features/organizations/hooks/useMemberships';
 
 const AllEventsList: FC = () => {
   const allEvents = useAllEvents();
+  const memberships = useMemberships().data;
```

Comment:
This hook is "old-school", in that it returns an `IFuture` instead of throwing a promise to make it compatible with Suspense.

Another problem with this hook is that it only returns "official memberships", i.e. memberships where the user's role is either admin or organizer.

The "membership" hooks are a mess, and we need to clean them up at some point.

For now, I will replace this with getting the organizations from the events instead, so that we allow filtering only on organizations that actually have events.